### PR TITLE
PIM-7209: write some scenarios for the catalog volume monitoring screen

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**

What I understood by talking with POs:

**Needs**

1. Administrator

Customers can have performance issues regarding the volume of the different entities in the PIM.
In order to guarantee the PIM performance, we are implementing a screen for the user administrator to follow the catalog volume. It will warn the user if a limit has been passed.

2. Maintenance

Our maintenance squad is interested by the volume of the catalog. 
But this is a different need, as there are much more information needed: version, OS, bundles, etc

3. Akeneo monitoring (Kibana)

Gathering statistics about the volume of the catalogs is not the same need as for the to previous point. The goal is to analyse what's the most common volume of our customers, in order to better optimize the PIM.

This PR just describes the first need, as it's the main goal of our subject.
It will not be merged as it is: it is just to be really aligned on what we need.

Please tell me if I did something wrong about gherkins.

Thanks @aRn0D that helped to write the scenarios 💟 

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    And the report returns that the mean number of attributes per family is 10
+    And the report returns that the maximum number of attributes per family is 10
+
+  Scenario: Warn the user administrator when the number of attributes per family has passed the limit
+    Given a families with 10 attributes
+    And a family with 4 attributes
+    And the limit of the maximum number of attributes per family is set to 8
+    And the limit of the mean number of attributes per family is set to 5
+    When the administrator user asks for the catalog volume monitoring report
+    Then the report returns warn the users that the maximum number of attributes per family has been passed
```
Comment (tamarasaurus) [inline tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: you could say "Then the report warns the user..." or "the report returns a warning that the ..."

File: tests/features/scalability/volume-monitoring/monitor_attribute_volume.feature [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    And 4 scopable attributes
+    And 3 localizable and scopable attributes
+    And 2 attributes neither localizable nor scopable
+    And the limit of the number of localizable attributes is set to 4
+    And the limit of the number of scopable attributes is set to 3
+    And the limit of the number of localizable and scopable attributes is set to 2
+    And the limit of the total number of attributes is set to 2
+    When the administrator user asks for the catalog volume monitoring report
+    Then the report returns warn the users that the total number of attributes has been passed
+    And the report returns warn the users that the number of localizable attributes has been passed
```
Comment (tamarasaurus) [inline tests/features/scalability/volume-monitoring/monitor_attribute_volume.feature]: limit has been passed?

File: tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Feature: Monitor catalog volume
+  In order to guarantee the performance of the PIM
+  As an administrator user
+  I want to monitor the volume of attributes per family
+
+  Scenario: Monitor the number of attributes per family
+    Given a families with 10 attributes
+    And a family with 4 attributes
+    When the administrator user asks for the catalog volume monitoring report
+    And the report returns that the mean number of attributes per family is 10
```
Comment (tamarasaurus) [inline tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: should it be Then ?

File: tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    And the report returns that the maximum number of attributes per family is 10
+
+  Scenario: Warn the user administrator when the number of attributes per family has passed the limit
+    Given a families with 10 attributes
+    And a family with 4 attributes
+    And the limit of the maximum number of attributes per family is set to 8
+    And the limit of the mean number of attributes per family is set to 5
+    When the administrator user asks for the catalog volume monitoring report
+    Then the report returns warn the users that the maximum number of attributes per family has been passed
+    Then the report returns warn the users that the mean number of attributes per family has been passed
```
Comment (aRn0D) [inline tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: And

File: tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  Scenario: Monitor the number of attributes per family
+    Given a families with 10 attributes
+    And a family with 4 attributes
+    When the administrator user asks for the catalog volume monitoring report
+    Then the report returns that the mean number of attributes per family is 7
+    And the report returns that the maximum number of attributes per family is 10
+
+  Scenario: Warn the user administrator when the number of attributes per family has passed the limit
+    Given a families with 10 attributes
```
Comment (NolwennP) [inline tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: a family

File: tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    And a family with 4 attributes
+    And the limit of the maximum number of attributes per family is set to 8
+    And the limit of the mean number of attributes per family is set to 5
+    When the administrator user asks for the catalog volume monitoring report
+    Then the report warns the users that the limit of maximum number of attributes per family has been passed
```
Comment (ahocquard) [inline tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: Then the report warns the users that the number of the attributes is high

Comment (jjanvier) [reply tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: I would have done 2 different tests for those 2 lines
there are 2 use cases here as far as I understand

max number of attributes reached
mean number of attributes reached

a test should always fail for one and only one reason

**Comment by jjanvier** [comment]
Cool to read some real Gherkin :+1: 

Did it help to communicate/understand the need?

File: tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature (hunk 6 above)
Comment (ahocquard) [reply tests/features/scalability/volume-monitoring/monitor_attribute_per_family_volume.feature]: @NolwennP checked this scenario and this is not the correct rule. There is actually only one rule: if your "mean" or your "max" is superior to the limit, you have a warning. As the max is always >= to the mean, the rule can be simplified this way:
```
 Scenario: Warn the user administrator when the maximum number of attributes per family is high
    Given a families with 8 attributes
    And a families with 2 attributes
    And the limit of the number of attributes per family is set to 6
    When the administrator user asks for the catalog volume monitoring report
    Then the report warns the users that the number of attributes per family is high
```

**Comment by ahocquard** [comment]
@jjanvier yes, it helped because I rewrote some parts and it highlighted some confusing rules.

**Pull request closed without merging**